Makefile

```diff
@@ -0,0 +1,122 @@
+STAGEDIR ?= "$(CURDIR)/stage"
+DESTDIR ?= "$(CURDIR)/install"
+ARCH ?= $(shell dpkg --print-architecture)
+SERIES ?= "bionic"
+
+ifeq ($(ARCH),arm64)
+	MKIMAGE_ARCH := "arm64"
+else ifeq ($(ARCH),armhf)
+	MKIMAGE_ARCH := "arm"
+else
+	$(error Build architecture is not supported)
+endif
+
+SERIES_HOST ?= $(shell lsb_release --codename --short)
+SOURCES_HOST ?= "/etc/apt/sources.list"
+SOURCES_MULTIVERSE := "$(STAGEDIR)/apt/multiverse.sources.list"
+
+# Download the latest version of package $1 for architecture $(ARCH), unpacking
+# it into $(STAGEDIR). For example, the following invocation will download the
+# latest version of u-boot-rpi for armhf, and unpack it under STAGEDIR:
+#
+#  $(call stage_package,u-boot-rpi)
+#
+define stage_package
+	mkdir -p $(STAGEDIR)/tmp
+	( \
+		cd $(STAGEDIR)/tmp && \
+		apt-get download \
+			-o APT::Architecture=$(ARCH) \
+			-o Dir::Etc::sourcelist=$(SOURCES_MULTIVERSE) $$( \
+				apt-cache \
+					-o APT::Architecture=$(ARCH) \
+					-o Dir::Etc::sourcelist=$(SOURCES_MULTIVERSE) \
+					showpkg $(1) | \
+					sed -n -e 's/^Package: *//p' | \
+					sort -V | tail -1 \
+			); \
+	)
+	dpkg-deb --extract $$(ls $(STAGEDIR)/tmp/$(1)*.deb | tail -1) $(STAGEDIR)
+endef
+
+
+classic: firmware uboot boot-classic device-trees gadget
+
+core: firmware uboot boot-core device-trees gadget
```

Comment:
Another thing that I like. This is how my original classic gadget Makefile looked in the past. It got reverted to one 'flat' makefile then because segmentation made no sense, as we were anyway building all targets (since the gadget was only for classic). Nice to finally see it re-added. It makes the whole thing look very clean.